scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -360,10 +363,14 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
             "Running $scannerDetails on directory '${pkgDownloadDirectory.absolutePath}'."
         }
 
-        archiveFiles(pkgDownloadDirectory, pkg.id, provenance)
+        if (provenance is KnownProvenance) {
+            archiveFiles(pkgDownloadDirectory, pkg.id, provenance)
+        }
 
         val (scanSummary, scanDuration) = measureTimedValue {
-            val vcsPath = provenance.vcsInfo?.takeUnless { it.type == VcsType.GIT_REPO }?.path.orEmpty()
+            val vcsPath = (provenance as? RepositoryProvenance)?.vcsInfo?.takeUnless {
+                it.type != VcsType.GIT_REPO
```

Comment:
This needs to stay `it.type == VcsType.GIT_REPO` like in the original (it's take*Unless*, not take*If*).